DNN-OpenCV-Classification-with-Java/README.md

```diff
@@ -0,0 +1,197 @@
+sudo apt-get install libatlas-base-dev gfortran
+sudo apt-get install python3-dev
+```
+
+For OpenCV Java installation we used default Java Runtime Environment and Java Development Kit:
+
+```bash
+sudo apt-get install default-jre
+sudo apt-get install default-jdk
+sudo apt-get install ant
+```
+
+2. Download the latest OpenCV version from the official repository:
+
+```bash
+cd ~
+wget -O opencv.zip https://github.com/opencv/opencv/archive/4.3.0.zip
+```
+
+3. Unzip the downloaded archives:
+
+```bash
+unzip opencv.zip
+```
+
+4. Rename the directories to match CMake paths:
+
+```bash
+mv opencv-4.3.0 opencv
+```
+
+5. Compile OpenCV. Create and enter a build directory:
+
+```bash
+cd ~/opencv
+mkdir build
+```
+
+6. Run CMake to configure the OpenCV build. Don't forget to set the right pass to the ``PYTHON_EXECUTABLE``:
+
+```bash
+cmake -D CMAKE_BUILD_TYPE=RELEASE \
+ -D CMAKE_INSTALL_PREFIX=/usr/local \
+ -D INSTALL_PYTHON_EXAMPLES=OFF \
+ -D INSTALL_C_EXAMPLES=OFF \
+ -D OPENCV_ENABLE_NONFREE=ON \
+ -D OPENCV_EXTRA_MODULES_PATH=~/opencv_contrib/modules \
+ -D PYTHON_EXECUTABLE=~/env/bin/python3 \
+ -D BUILD_SHARED_LIBRARY=OFF \
+ -D BUILD_TESTS=OFF \
+ -D BUILD_PERF_TESTS=OFF \
+ -D BUILD_EXAMPLES=ON ..
+```
+
+If you want to configure the build with some specific Java version, please, add the following fields, verifying the paths:
+
+```bash
+ -D JAVA_AWT_INCLUDE_PATH=/usr/lib/jvm/java-1.x.x-openjdk-amd64/include \
+ -D JAVA_AWT_LIBRARY=/usr/lib/jvm/java-1.x.x-openjdk-amd64/lib/libawt.so \
+ -D JAVA_INCLUDE_PATH=/usr/lib/jvm/java-1.x.x-openjdk-amd64/include \
+ -D JAVA_INCLUDE_PATH2=/usr/lib/jvm/java-1.x.x-openjdk-amd64/include/linux \
+ -D JAVA_JVM_LIBRARY=/usr/lib/jvm/java-1.x.x-openjdk-amd64/include/jni.h \
+```
+
+7. Check the output and make sure that everything is set correctly. After that we're ready to build it with:
+
+```bash
+make -j8
+```
+
+Make sure, you didn't get any errors. In case of successful completion you will find the following files in the ``build`` directory:
+* ``bin/opencv-430.jar``
+* ``lib/libopencv_java430.so``
+* ``lib/python3/cv2.cpython-37m-x86_64-linux-gnu.so``
+
+Then run the following command:
+
+```bash
+sudo ldconfig
+```
+
+which creates the necessary links and cache to our freshly built shared library.
+
+Rename the created Python3 bindings for OpenCV to `cv2.so`:
+
+```bash
+mv lib/python3/cv2.cpython-37m-x86_64-linux-gnu.so cv2.so
+```
+
+Create a symlink of our OpenCV `cv2.so` into the virtual environment installed packages:
+
+```bash
+cd env/lib/python3.7/site-packages/
+ln -s ~/opencv/build/cv2.so cv2.so
+```
+
+The last step is to put ``~/opencv/build/lib/libopencv_java430.so`` to the ``/usr/lib`` directory:
+
+```bash
+sudo cp lib/libopencv_java430.so /usr/lib
+```
+
+For Windows and macOS OpenCV Java build, please, follow the steps described in [Installing OpenCV for Java](https://opencv-java-tutorials.readthedocs.io/en/latest/01-installing-opencv-for-java.html).
```

Comment:
There is official tutorial about OpenCV Java bindings: https://docs.opencv.org/master/d9/d52/tutorial_java_dev_intro.html